### PR TITLE
Ignore all build folders in .gitignore

### PR DESCRIPTION
**Pull request by deuteragenie**
I suppose we don't want any build* folder ending up in git, so ignore any folder starting with build instead of only build32 and build64.  This is consistent with semaphore which uses build.gcc and build.clang for example.

### CODE REVIEW

**Review by Lord-Kamina — APPROVED** [review]
Makes sense to me, at least.

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,4 @@
-build
-build32
-build64
+build*
```
Comment (Baklap4) [inline .gitignore:1]: i'd rather use:
`[bB]uild*`

Also on visual studio the default is an `out` folder.

**Review by Baklap4 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (Tronic) [reply .gitignore:1]: I'd suggest `/[bB]uild*/` with the first slash making it match only repository root and the final slash making it match folders only.

Comment (deuteragenie) [reply .gitignore]: Done.  I guess this can be merged into master now.

**Pull request merged**